### DISCUSSION TITLE
v5.54.0

### DISCUSSION THREAD

**Discussion by sokra** [webpack/webpack#14323] [General]
# Features

* improve constant folding to allow to skip more branches for `&&` `||` and `??`
* allow all hashing using in webpack to be configured with `output.hashFunction`
* no longer bailout completely from inner graph analysis when `eval` is used in a module

# Bugfixes

* force bump enhanced-resolve for bugfixes

# Performance

* reduce number of allocation when creating snapshots
* add `output.hashFunction: "xxhash64"` for a super fast wasm based hash function
* improve utf-8 conversion when serializing short strings
* improve hashing performance for dependencies
* add `experiments.cacheUnaffected` which caches computations for modules that are unchanged and reference only unchanged modules

<hr /><em>This discussion was created from the release <a href='https://github.com/webpack/webpack/releases/tag/v5.54.0'>v5.54.0</a>.</em>

**Comment by alan-agius4** [comment]
@sokra, can you please expand on `cacheUnaffected`. I am having some trouble understanding that it really does. From the above description and documentation.

**Reply by sokra** [reply]
It adds additional caching for modules that are unaffected by the changes.

We invalid these cache only when the source code has changed or when imported dependencies have changed (potentially transitively when reexports are used).

This allows to cache some computations during sealing, e.g. hashing, exports computation, runtime requirements. In future also parts of the chunk graph generation.

This improves incremental build performance.

**Reply by alan-agius4** [reply]
Thanks for the clarification Tobias!
